sdk/provisioning/Azure.Provisioning/src/Parameter.cs

```diff
@@ -28,9 +28,13 @@ namespace Azure.Provisioning
         public bool IsSecure { get; }
         internal bool IsFromOutput { get; }
         internal bool IsLiteral { get; }
-        internal string? Value { get; }
-        internal IConstruct? Source { get; }
+        internal string? Value { get; set; }
+        internal IConstruct? Source { get; set; }
```

Comment:
I am slightly concerned that making this mutable could lead to inconsistencies.  Would create a new Parameter from the old make more sense like we did with Output?

Or perhaps a SynthesizedParameter similar to the SynthesizedResource suggestion?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/42035